MyPerfectOnboardingApplication/Tests/MyPerfectOnboarding.App.Tests/Api/Controllers/ListControllerApplicationTests.cs

```diff
@@ -0,0 +1,282 @@
+            });
+        }
+
+        [Test]
+        public async Task DeleteItem_SameItemTwice_DeleteItemAndSecondTimeReturnNotFound()
+        {
+            var itemIdToDelete = _items[1].Id;
+            _listRepository.GetAllItemsAsync().Returns(_items);
+
+            var message1 = await _listController.ExecuteAction(controller => controller.DeleteAsync(itemIdToDelete));
+            var message2 = await _listController.ExecuteAction(controller => controller.DeleteAsync(itemIdToDelete));
+
+            Assert.Multiple(() => {
+                Assert.That(message1.StatusCode, Is.EqualTo(HttpStatusCode.NoContent));
+                Assert.That(message2.StatusCode, Is.EqualTo(HttpStatusCode.NotFound));
+
+                _listRepository.Received(1).GetAllItemsAsync();
+                _listRepository.Received(1).DeleteItemAsync(Arg.Is<Guid>(id => id.Equals(itemIdToDelete)));
+            });
+        }
+
+        [Test]
+        public async Task AddItemAndGetAllItems_Item_ReturnsAllItemsIncludedNewOne()
+        {
+            _listRepository.GetAllItemsAsync().Returns(_items);
+
+            var stringId = "22AC59B7-9517-4EDD-9DDD-EB418A7C1678";
+            var id = Guid.Parse(stringId);
+            var expectedUri = new Uri($"http://www.aaa.com/{stringId}");
+            _location.GetListItemLocation(id).Returns(expectedUri);
+
+            var stringTime = "2150-12-05";
+            var addedItemCreationTime = DateTime.Parse(stringTime);
+
+            _timeGenerator.GetCurrentTime().Returns(addedItemCreationTime, DateTime.MinValue);
+            _guidGenerator.Generate().Returns(id);
+
+            var newItemText = "newItemText";
+            var newItem = ListItemViewModelBuilder.CreateItem(newItemText);
+            var addedItem = ListItemBuilder.CreateItem(stringId, newItemText, stringTime);
+            _listRepository.AddItemAsync(ArgExtended.IsListItem(addedItem)).Returns(addedItem);
+            var newItemsList = _items.Append(addedItem);
+
+            var message1 = await _listController.ExecuteAction(controller => controller.PostAsync(newItem));
+            message1.TryGetContentValue(out IListItem resultItem);
+            var message2 = await _listController.ExecuteAction(controller => controller.GetAsync());
+            message2.TryGetContentValue(out IEnumerable<IListItem> resultItems);
+
+            Assert.Multiple(() => {
+                Assert.That(message1.StatusCode, Is.EqualTo(HttpStatusCode.Created));
+                Assert.That(message2.StatusCode, Is.EqualTo(HttpStatusCode.OK));
+                Assert.That(resultItem, Is.EqualTo(addedItem).UsingListItemComparer());
+                Assert.That(resultItems, Is.EquivalentTo(newItemsList).UsingListItemComparer());
+
+                _listRepository.Received(1).GetAllItemsAsync();
+            });
+        }
+
+        [Test]
+        public async Task GetItemReplaceItemGetItOnceMore_IdAndItem_ReturnsEditedItem()
```

Comment:
I'm mostly wondering what this "middle" name in your application-wide tests means (I mean, it is OK to have a name separated by multiple underscores, however, I don't get the meaning in these. Should it be something like
```suggestion
        public async Task GetItem_ReplaceItem_GetItOnceMore_ReturnsEditedItem()
```
?